modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/jobs/VersionMigrationsWorker.kt

```diff
@@ -74,14 +86,9 @@ class VersionMigrationsJob : JobService() {
             }
             Timber.i("Migrating from version $previousVersionCode to $versionCode")
 
-            LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "Running VersionMigrationsTask")
-            val jobScheduler = context.getSystemService(Context.JOB_SCHEDULER_SERVICE) as JobScheduler
-            jobScheduler.cancel(JobIds.VERSION_MIGRATION_JOB_ID)
-            jobScheduler.schedule(
-                JobInfo.Builder(JobIds.VERSION_MIGRATION_JOB_ID, ComponentName(context, VersionMigrationsJob::class.java))
-                    .setOverrideDeadline(500) // don't let Android wait for more than 500ms before kicking this off
-                    .build(),
-            )
+            LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "Running VersionMigrationsWorker")
+            val workRequest = OneTimeWorkRequestBuilder<VersionMigrationsWorker>().build()
+            WorkManager.getInstance(context).enqueueUniqueWork(VERSION_MIGRATIONS_WORKER_NAME, ExistingWorkPolicy.REPLACE, workRequest)
```

Comment:
I think `KEEP` is a more appropriate policy. If the job gets canceled midway after updating migrated version code it will skip all migrations.

Reply:
Makes sense, replaced in f609a54.